src/core/annotation.js

```diff
@@ -1110,6 +1109,61 @@ class TextAnnotation extends MarkupAnnotation {
   }
 }
 
+class FreeTextAnnotation extends Annotation {
+  constructor(parameters) {
+    super(parameters);
+
+    this.data.annotationType = AnnotationType.FREETEXT;
+    const dict = parameters.dict;
+    this.pdfManager = parameters.pdfManager;
+
+    this.data.textColor = createRgbColor(dict.getArray("TextColor"));
+    this.data.textStyle = dict.get("DS");
+    this.data.fontSize = dict.get("FontSize");
+    this.data.opacity = dict.get("CA");
+    this.data.defaultAppearance = dict.get("DA");
+  }
+
+  getOperatorList(evaluator, task, renderForms) {
+    if (renderForms || this.appearance) {
+      return super.getOperatorList(evaluator, task, renderForms);
+    }
+
+    const data = this.data;
+
+    const opList = new OperatorList();
+    const appearanceStream = new Stream(stringToBytes(data.defaultAppearance));
+    const resourcesFonts = this.pdfManager.pdfDocument.catalog.fontCache.dict;
```

Comment:
Unfortunately https://github.com/mozilla/pdf.js/pull/10122#discussion_r224351991 still applies here, there's simply no way that this works correctly.

Reply:
I tried resolving this in https://github.com/mozilla/pdf.js/pull/12093/commits/e9ed30cbe958a9aa430636443a628f769c96f79c . Does that look sane?